core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,562 @@
+    outfile = osp.join(tmp_dir, "mutations.maf")
+    # Fill empty cells with NA string, otherwise they get parsed incorrectly by SigProfilerMatrixGenerator
+    maf.to_csv(outfile, sep="\t", index=False, na_rep="NA")
+    return outfile, ncbi_build
+
+
+def preprocess_vcf(args: argparse.Namespace, tmp_dir):
+    """Preprocess vcf files by copying them into the tmp dir
+    
+    :param args: argparse parsed arguments
+    :param tmp_dir: temporary directory
+    """
+    vcf_list = glob.glob(osp.join(args.vcf_folder, "*.vcf"))
+    for vcf in vcf_list:
+        shutil.copy(vcf, osp.join(tmp_dir, osp.basename(vcf)))
+
+
+def prepare_input(args: argparse.Namespace) -> tuple:
+    """Wrapper function for preparing input files for matrix generation
+    
+    :param args: argparse parsed arguments 
+    :return: ncbi_build, study ID
+    """
+    # Init variables
+    maf_file, study_id, ncbi_build = [""] * 3
+    
+    if args.study_path:
+        # Get maf file and study ID from cBioPortal study metadata
+        maf_file, study_id = acquire_maf_study_id(args.study_path)
+    elif args.maf:
+        maf_file = args.maf
+
+    if maf_file:
+        # Check if given maf file exists in study path, fix order of columns and copy to temporary directory
+        maf_file, ncbi_build = preprocess_maf(maf_file, args.tmp_dir, args)
+    else:
+        # No MAF, so VCF as input
+        preprocess_vcf(args, args.tmp_dir)
+    
+    # Set ncbi_build and study_id if provided as argument
+    if args.ncbi_build:
+        ncbi_build = args.ncbi_build
+    if args.study_id:
+        study_id = args.study_id
+    
+    return ncbi_build, study_id
+    
+
+def genome_install(ncbi_build: str):
+    """Install a genome for SigProfilerMatrixGenerator
+    
+    :param ncbi_build: ncbi build
+    """
+    print(f"Installing genome: {ncbi_build} for SigProfilerMatrixGenerator.")
+    genInstall.install(ncbi_build)
+
+
+def run_matrix_generator(ncbi_build: str, input_folder: str):
+    """Run SigProfilerMatrixGenerator to generate substitution matrices in input folder
+    
+    :param ncbi_build: ncbi build
+    :param input_folder: path to folder containing maf/vcf input files
+    """
+    print("-" * 120 + "\nGenerating nucleotide matrices: ")
+    # Name of the project is irrelevant, the name will not be used any further
+    project = "tmp"
+    matGen.SigProfilerMatrixGeneratorFunc(project, ncbi_build, input_folder)
+
+
+def matrix_stable_id(string: str) -> str:
+    """Replace non-stableID characters by - and _ in matrix names
+    
+    :param string: matrix name
+    :return: stable ID with only allowed characters
+    """
+    replace = {
+        ">": "-", ":": "_", "[": "_", "]": "_"
+    }
+    new = ["mutational_signatures_matrix_"]
+    for char in string:
+        if replace.get(char):
+            new.append(replace.get(char))
+        else:
+            new.append(char)
+    return "".join(new)
```

Comment:
is easier resolved using regex
```suggestion
import re

string = re.sub(r'>', '-', string)
string = re.sub(r'[\[\]:]', '_', string)
return string
```

Reply:
of course, put the `import re` to the opening of the script

Reply:
Nice clean solution, implemented